09-writing-advanced-plans/README.md

```diff
@@ -46,20 +46,26 @@ In the previous exercise you ran tasks and commands within the context of a plan
 3. Create a plan and save it as `modules/exercise9/plans/yesorno.pp`:
 
     ```puppet
+    # TargetSpec accepts a comma-separated list of nodes
     plan exercise9::yesorno (TargetSpec $nodes) {
+      # Run the 'exercise9::yesorno' task on the nodes you specify.
       $results = run_task('exercise9::yesorno', $nodes)
-      $subset = $results.filter |$result| { $result[answer] == true }.map |$result| { $result.target }
-      run_command("uptime", $subset)
+
+      # Puppet uses immutable variables. That means we have to operate on data, in
+      # this case a ResultSet containing a list of Results. Those are documented in
+      # Bolt's docs, but effectively its a list of result data parsed from JSON
+      # objects returned by the task. Collect - "filter" - only the Result objects
```

Comment:
We should change "collect" to "select" to avoid confusion for anyone familiar with `collect` in Ruby (which is map)